core/src/sumeragi/main_loop.rs

```diff
@@ -1065,6 +1100,15 @@ fn vote_for_block<F: FaultInjection>(
         return None;
     }
 
+    if block.header().committed_with_topology != state.current_topology {
+        error!(
+            %addr, %role, block_topology=?block.header().committed_with_topology, my_topology=?state.current_topology, hash=%block.hash(),
```

Comment:
```suggestion
            %addr, %role, block_topology=%block.header().committed_with_topology, my_topology=%state.current_topology, hash=%block.hash(),
```

Consider implementing `core::fmt::Display` for `block_header` and `Topology`.

Reply:
I don't think it would be useful.